go-crud/books/handler.go

```diff
@@ -0,0 +1,182 @@
+package books
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/supercede/go-exercises/go-crud/models"
+
+	"github.com/supercede/go-exercises/go-crud/data"
+)
+
+type Handler struct {
+	store *data.Store
+}
+
+func NewHandler(s *data.Store) *Handler {
+	return &Handler{store: s}
+}
+
+func (h *Handler) createBook(w http.ResponseWriter, r *http.Request) {
+	store := h.store
+
+	err := store.ReadFromFile()
+
+	if err != nil {
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var b models.Book
+
+	err = json.NewDecoder(r.Body).Decode(&b)
+	if err != nil {
+		if err == io.EOF {
+			http.Error(w, "Empty Request Body", http.StatusBadRequest)
+			return
+		}
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	_, err = validateBook(b)
+
```

Comment:
nit: You don't need these empty lines between error checking
```go
	_, err = validateBook(b)
if err != nil {
...
}
```
Is more idiomatic.